crime-means-assessment/src/main/java/uk/gov/justice/laa/crime/meansassessment/service/stateless/StatelessAssessmentService.java

```diff
@@ -21,41 +24,67 @@
 
 import static uk.gov.justice.laa.crime.meansassessment.service.stateless.DataAdapter.convertChildGroupings;
 
-@RequiredArgsConstructor
 @Service
 public class StatelessAssessmentService extends BaseMeansAssessmentService {
+
     private final AssessmentCriteriaService assessmentCriteriaService;
-    private final AssessmentCriteriaChildWeightingService childWeightingService;
+    private final MeansAssessmentServiceFactory meansAssessmentServiceFactory;
+    private final FullAssessmentAvailabilityService fullAssessmentAvailabilityService;
+
+    public StatelessAssessmentService(AssessmentCriteriaService assessmentCriteriaService,
+                                      MeansAssessmentServiceFactory meansAssessmentServiceFactory,
+                                      FullAssessmentAvailabilityService fullAssessmentAvailabilityService) {
+
+        super(assessmentCriteriaService);
+        this.assessmentCriteriaService = assessmentCriteriaService;
+        this.meansAssessmentServiceFactory = meansAssessmentServiceFactory;
+        this.fullAssessmentAvailabilityService = fullAssessmentAvailabilityService;
+    }
+
+    public StatelessResult execute(Assessment assessment,
+                                   Map<AgeRange, Integer> childGroupings,
+                                   List<Income> income,
+                                   List<Outgoing> outgoings) {
 
-    public StatelessResult invoke(Assessment assessment,  Map<AgeRange, Integer> childGroupings, List<Income> income, List<Outgoing> outgoings) {
         if (assessment.getAssessmentType() == StatelessRequestType.INITIAL) {
-            return initialOnly(assessment.getAssessmentDate(), assessment.getHasPartner(),
-                    assessment.getCaseType(), assessment.getMagistrateCourtOutcome(),
-                    childGroupings, income);
+            return initialOnly(
```

Comment:
Can we use some input classes for these methods? We've got up to 8 inputs in some and most appear to be common so there must be a way we can consolidate?